src/Parser.hs

```diff
@@ -0,0 +1,310 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Parser where
+
+import Text.Megaparsec
+    ( Parsec
+    , MonadParsec (lookAhead, hidden, eof, takeWhile1P, try)
+    , choice
+    , skipSome
+    , between
+    , (<?>)
+    )
+import Data.Void (Void)
+import Lib
+    ( Primitive (Boolean, Constant, SymbolReference, SymbolList)
+    , Symbol (Symbol)
+    , Expression (Operation, Primitive)
+    , ifOperator
+    , Arguments (Triple, Single, Pair, List)
+    , Operator (Unary, Binary, Ternary, Nary)
+    , defineOperator
+    , divideOperator
+    , multiplyOperator
+    , subtractOperator
+    , addOperator
+    , oppositeOperator
+    , notOperator
+    , moduloOperator
+    , eqOperator
+    , inferiorOperator
+    , callOperator
+    , lambdaOperator
+    )
+import Text.Megaparsec.Char (space1, char)
+import Data.Functor (($>), void, (<&>))
+import Control.Applicative ((<|>), Alternative (many))
+import qualified Text.Megaparsec.Char.Lexer as L
+import Text.Megaparsec.Char.Lexer (signed)
+import Data.Text (Text, unpack, uncons, all, splitAt)
+import Data.Char
+    ( isSpace
+    , generalCategory
+    , GeneralCategory
+        ( UppercaseLetter
+        , LowercaseLetter
+        , TitlecaseLetter
+        , ModifierLetter
+        , OtherLetter
+        , NonSpacingMark
+        , LetterNumber
+        , OtherNumber
+        , DashPunctuation
+        , OtherPunctuation
+        , CurrencySymbol
+        , MathSymbol
+        , ModifierSymbol
+        , OtherSymbol
+        , PrivateUse
+        , DecimalNumber
+        , SpacingCombiningMark
+        , EnclosingMark
+        )
+    )
```

Comment:
I think we should have a consistent style for the imports, so either format `Lib.hs` like this or those imports the same way as in `Lib.hs`

Reply:
I completely agree. I was using the "leading-comma" style because I thought import lists were an other case where haskell doesn't allow a trailing comma (past the last element), which is the style I usually prefer, because it makes better diffs when adding or removing elements. For that purpose, I'll change it to be one element per line, with a trailing comma, or all elements on a single line if there are like 5 or less elements in the entire list

I'll also make the change in Lib for consistency